File: centre-ui/src/app/services/models/parent-detail-response.ts
/* tslint:disable */
/* eslint-disable */
import { StudentResponse } from '../models/student-response';
export interface ParentDetailResponse {
  email?: string;
  firstName?: string;
  id?: number;
  isDeceased?: boolean;
  lastName?: string;
  maritalStatus?: 'MARRIED' | 'DIVORCED' | 'WIDOWED';
  phoneNumber?: string;
  profession?: string;
  students?: Array<StudentResponse>;
  type?: 'FATHER' | 'MOTHER';
}

File: centre-api/src/main/java/com/centre/poly/person/dto/ParentDetailResponse.java
package com.centre.poly.person.dto;

import com.centre.poly.person.entity.MaritalStatus;
import com.centre.poly.person.entity.ParentType;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class ParentDetailResponse {
    
    private Long id;
    private String firstName;
    private String lastName;
    private String phoneNumber;
    private String email;
    private ParentType type;
    private String profession;
    private MaritalStatus maritalStatus;
    private Boolean isDeceased;
    private List<StudentResponse> students;
    
}
